File: rwebspec/pages/flight_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class FlightPage < AbstractPage

  def initialize(driver)
    super(driver, "") # <= TEXT UNIQUE TO THIS PAGE
  end

  def select_trip_type(trip_type)
    click_radio_option("tripType", trip_type)
  end

  def select_depart_from(from_port)
    select_option("fromPort", from_port)
  end

  def select_arrive_at(to_port)
    select_option("toPort", to_port)
  end

  def select_depart_day(depart_day)
    select_option("departDay", depart_day)
  end

  def select_depart_month(depart_month)
    select_option("departMonth",depart_month)
  end

  def select_return_day(return_day)
    select_option("returnDay", return_day)
  end

  def select_return_month(return_month)
    select_option("returnMonth", return_month)
  end

  def click_continue
    click_button("Continue")
  end

end

File: playwright-mocha-chrome/pages/flight_page.js
const AbstractPage = require('./abstract_page');

class FlightPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async selectTripType(trip_type) {
    const radios = await this.driver.$$("input[name=tripType]");
    if (trip_type == "oneway") {
  await radios[1].check();
    } else  {
        await radios[0].check();
    }
  }

  async selectDepartFrom(city) {
    await this.driver.selectOption("select[name='fromPort']", city);
  }

  async selectArriveAt(city) {
    await this.driver.selectOption("select[name='toPort']", city);
  }

  async selectDepartDay(day) {
    await this.driver.selectOption("select[name='departDay']", day);
  }

  async selectDepartMonth(month_year) {
     await this.driver.selectOption("#departMonth", month_year);
  }

  async selectReturnDay(day) {
    await this.driver.selectOption("select[name='returnDay']", day);

  }

  async selectReturnMonth(month_year) {
    await this.driver.selectOption("#returnMonth", month_year);
  }

  async clickContinue() {
    await this.driver.click("input:has-text('Continue')");
  }
};

module.exports = FlightPage;
